File: archive/m/matlab/prime_number.m
function result = prime_number(n)

if n < 0
	result = 'Please input a non-negative integer'
end

if not(ischar(n))
	result = 'Please input a non-negative integer'
end

if not(isempty(n)) = 1
	result = 'Please input a non-negative integer'
end

if not(~isinf(x) & floor(x) == x)
    result = 'Please input a non-negative integer'
end

if n == 0 | n == 1
    result = 'Neither Prime nor Composite'
end

m = 2;
isprime = 1;

while (m <= sqrt(n))
    if(rem(n, m) == 0)
        isprime = 0;
        break;
    end
    m = m + 1;
end

if(isprime == 1)
    result = 'Prime'
else
    result = 'Composite'

end

File: archive/d/dart/prime-number.dart
// Issue 4970
import 'dart:math';

void main(List<String> args){
  const String error_message= "Usage: please input a non-negative integer";
  if (args.isEmpty){
    print(error_message);
    return;
  }

  try{
    bool is_prime = true;
    int num_needed = int.parse(args[0]);

    if (num_needed.isNegative){
      print(error_message);
      return;
    }

    // If the number is even number other than 2 OR if the number is multiple of 5 other than 5 or if number is either 0 or 1, Print Composite
    if( (num_needed == 0)|| (num_needed == 1) || (num_needed %2 ==0 && num_needed != 2) || (num_needed %5  ==0 && num_needed != 5) ){
      print("Composite");
      return;
    }

    for (int i_index = 3; i_index <= sqrt(num_needed).toInt(); i_index +=2){
      if(num_needed % i_index == 0){
        is_prime = false;
        break;
      }
    }

    print(is_prime ? "Prime" : "Composite");
  }
  catch(e){
    print(error_message);
  }
}
